### ISSUE TITLE
[Day 4] CNN으로 CIFAR-10 분류 완료 - {이름}

### ISSUE THREAD

**Issue by J00NQ** [J00NQ/algorithm_YH#10]
## 실습 완료 내용

- [x] Kaggle Learn Lesson 1~3 완료 (The Convolutional Classifier, Convolution and ReLU, Maximum Pooling)

- [x] CIFAR-10 데이터 로드 & 탐색

- [x] CNN 모델 구축 (Conv2D + MaxPooling + Dense)

- [x] 모델 훈련 (조기 종료 적용)

- [x] 성능 평가 및 혼동행렬 분석

- [x] Dense Network와 성능 비교

## 데이터셋 정보

- 이미지 크기: 32×32 (컬러)

- 훈련 데이터: 40,000장

- 검증 데이터: 10,000장

- 테스트 데이터: 10,000장

- 클래스: 10가지 (airplane, automobile, bird, ..., truck)

## 모델 구축

- Conv2D(32, 3×3) + ReLU + MaxPooling(2×2)

- Conv2D(64, 3×3) + ReLU + MaxPooling(2×2)

- Conv2D(128, 3×3) + ReLU + MaxPooling(2×2)

- Flatten()

- Dense(256) + ReLU + Dropout(0.5)

- Dense(10) + Softmax (10-class 분류)

## 성능 결과

- 훈련 정확도: 87.42% 

- 검증 정확도: 74.25% 

- 테스트 정확도: 74.17% (최종)

- 총 파라미터: 620,362개

- 훈련 에포크: 11개 (조기 종료 적용)

## Dense Network와의 비교

| 항목 | Dense | CNN |

|------|-------|-----|

| 입력 형태 | 1D 벡터 | 2D 이미지 |

| 파라미터 수 | 약 77M | 약 0.6M개 |

| 정확도 | 70~78% | 70~78% |

| 훈련 시간 | 2~5분 | 2분 |

| 메모리 | ~2GB | 500MB |

## 관찰 기록

1. CNN이 Dense Network보다 효율적인 이유:

   - 파라미터 감소: CNN이 약 125배 더 효율적!
   
   - 공간 정보 보존: Dense: 이미지를 1D 벡터로 변환 → 공간 정보 손실, CNN: 2D 구조 유지 → 인접 픽셀 관계 학습

   - 특징 학습의 계층성: Dense: 전체 픽셀을 함께 처리, CNN: 지역 패턴부터 시작 (엣지→텍스처→객체)

2. 가장 잘 분류되는 클래스 Top 3:

   - truck: 0.8690 (86.90%)

   - frog: 0.8570 (85.70%)

   - ship: 0.8470 (84.70%)

3. 가장 어려운 클래스 (정확도 낮음):

   - cat: 0.5410 (54.10%) → 이유: dog과의 혼동

   - dog: 0.6260 (62.60%) → 이유: cat과의 혼동

4. 가장 많이 혼동되는 클래스 쌍:

   - cat → dog (179건)

   - dog → cat (166건)

## 개선 시도 (선택)

- 시도 1: (예: 필터 개수 증가) → 결과: ____%

- 시도 2: (예: Batch Normalization 추가) → 결과: ____%

- 가장 효과 있었던 변경: 

## 다음 단계 (내일)

Day 5: YOLO로 객체 검출 도전 예정

## 커밋 및 링크

-[ Kaggle 노트북: (URL)](https://www.kaggle.com/code/j00n99/day04)

- GitHub: (링크)

**Comment by J00NQ** [comment]
<img width="389" height="789" alt="Image" src="https://github.com/user-attachments/assets/48607baf-51b0-473e-9b13-800569737c2a" />

**Comment by J00NQ** [comment]
<img width="639" height="622" alt="Image" src="https://github.com/user-attachments/assets/2383df7d-6790-40a0-b4af-bdd64720bb28" />

**Comment by J00NQ** [comment]
<img width="909" height="755" alt="Image" src="https://github.com/user-attachments/assets/8403b746-4db0-4f20-9f73-7f9d5b933741" />

**Comment by J00NQ** [comment]
<img width="1197" height="394" alt="Image" src="https://github.com/user-attachments/assets/ce001225-44b8-4ddc-8d33-23120525ba3e" />

<img width="589" height="272" alt="Image" src="https://github.com/user-attachments/assets/850af5a1-7c73-4c6e-bc71-052c9e3bea01" />

<img width="323" height="116" alt="Image" src="https://github.com/user-attachments/assets/05793ebe-96e9-49d1-92a3-b21e4401e7f7" />

**Comment by J00NQ** [comment]
<img width="975" height="782" alt="Image" src="https://github.com/user-attachments/assets/f5b5bf28-f328-42f3-a412-e765fba3ffba" />

<img width="427" height="336" alt="Image" src="https://github.com/user-attachments/assets/1c994d02-791c-4b86-8f11-e7b1bc14de3b" />

<img width="243" height="220" alt="Image" src="https://github.com/user-attachments/assets/6f24b415-31f1-48e8-9cbc-a1c7110cca3e" />

<img width="620" height="664" alt="Image" src="https://github.com/user-attachments/assets/af379dbd-b4bb-40e2-8f00-768d15e37f07" />

**Comment by J00NQ** [comment]
<img width="452" height="411" alt="Image" src="https://github.com/user-attachments/assets/4d02ded7-383a-4542-92a6-568883efa7aa" />
